docs/docker.md

```diff
@@ -38,3 +38,113 @@ docker-compose up -d
 
 - Open [localhost:3000](http://localhost:3000/) for the server.
 - Open [localhost:5555](http://localhost:5555/) for the Prisma database browser.
+
+
+#### Checking the logs
+
+```bash
+docker-compose logs -f
+```
+
+#### Checking logs for a web service
+
+```bash
+docker-compose logs -f web 
+```
+or
+```bash
+docker logs placemark-web
+```
+
+#### Checking logs for a db service
+
+```bash
+docker-compose logs -f db 
+```
+or
+```bash
+docker logs placemark-db
+```
+
+#### Checking logs for a prisma studio service
+
+```bash
+docker-compose logs -f prisma 
+```
+or
+```bash
+docker logs placemark-studio
+```
+
+#### Stopping all the services
+
+```bash 
+docker-compose down
+```
+
+#### Stopping a specific service
+
+```bash 
+docker-compose stop prisma
+```
+
+#### checking the database
+
+```bash 
+docker-compose exec db psql -U postgres -d placemark
+``` 
+
+#### Running prisma studio
+
+```bash 
+docker-compose exec prisma npx prisma studio
+```
+
+#### Running prisma migrate
+
+```bash
+docker-compose exec prisma npx prisma migrate dev --name init
+```
+
+#### Running prisma generate
+
+```bash
+docker-compose exec prisma npx prisma generate
+```
+
+#### Running prisma introspect
+
+```bash
+docker-compose exec prisma npx prisma db pull
+```
+
+
+#### Terminal access to the web service
+
+```bash
+docker-compose exec web bash
+```
+
+#### Terminal access to the db service
+
+```bash 
+docker-compose exec db bash
+```
+
+#### Rebuilding and recreate the web service
+
+```bash
+docker-compose up -d --build --force-recreate web
+```
+#### Resarting the web service
+
+```bash
+docker-compose restart web
+```
+
+Note: We are not exposing the database to the host machine, so you can't connect to the database using a database client like pgAdmin or DBeaver. If you want to connect to the database, you can use the terminal access to the db service. If you want to expose the database to the host machine, you can add the following line to the db service in the docker-compose.yml file.
```

Comment:
Why not expose the database? This Docker file is for development, and afaict it's exposing the Prisma browser still, so why not let people access the database too? A prod Docker fille will have to be different in a bunch of ways.

Reply:
Yeah, as you have mentioned, in development, it's okay to have an open port. However, there are some useful cases where I personally prefer not to expose the port for the following two reasons in development as well:

1. Nowadays, people often have multiple PostgreSQL containers running on the same machine for different use cases. If we expose the port at 5432, we need to ensure that it is not being used by another container. Additionally, we would have to determine which port we want to allocate, for example, 5433, 5444, etc. Therefore, by disabling and connecting the container within the same network, as I have done, since the port is not exposed and containers communicate through the same network, there won't be any issues, making it easier to set up the project.

2. Similar to the first reason, here we have hardcoded the port as 5432. Since many people have PostgreSQL installed on their machines, port 5432 is likely already in use. To address this, I have adopted this approach so that the project can be set up properly. Later, if developers wish to expose it for any reason, they can make the necessary changes to the compose file.

Let me know if we do not want to follow this approach, and I will make the required changes

Reply:
Okay - got it, thanks for the detail!